### PR TITLE
Fix account check marks in filter menu

### PR DESCRIPTION
**Pull request by erikjv**
Fixes finding from:
- https://github.com/owncloud/client/issues/10884#issuecomment-1649818735

Also fixes interaction with the caller side of the method populating the menu.

### CODE REVIEW

**Comment by fmoc** [comment]
Please provde a meaningful title and a description. If there are UI changes, please add screenshots.

**Review by fmoc — CHANGES_REQUESTED** [review]
See above.

**Comment by erikjv** [comment]
The separate commits have a useful description, but I can put them in the PR.

File: src/gui/models/models.cpp [hunk 1]
```diff
@@ -99,9 +99,7 @@ std::function<void()> OCC::Models::addFilterMenuItems(QMenu *menu, const QString
             model->setFilterFixedStringSignalled(filter);
         });
         action->setCheckable(true);
-        if (currentFilter == filter) {
-            action->setChecked(true);
-        }
+        action->setChecked(currentFilter == QRegularExpression::escape(filter));
```
Comment (fmoc) [inline src/gui/models/models.cpp:102]: The logic change is not explained by the commit message. You could have compared the value to the escaped expression in the `if`, too.

**Review by fmoc — CHANGES_REQUESTED** [review]

File: src/gui/models/models.cpp (hunk 1 above)
Comment (TheOneRing) [reply src/gui/models/models.cpp:102]: Most probably Qt6 porting fallout.
The regex filter api changed for Qt6.

Comment (erikjv) [reply src/gui/models/models.cpp:102]: It's shorter and does the same thing.

Comment (fmoc) [reply src/gui/models/models.cpp:102]: > does the same thing

Well, not quite. We have never explicitly set `false` before. It apparently works fine in this case, but I wouldn't assume this as the default in general.

Anyway, good enough for me.

**Review by fmoc — APPROVED** [review]

**Pull request merged**